include/boost/url/authority_view.hpp

```diff
@@ -71,81 +70,15 @@ namespace urls {
 */
 class BOOST_SYMBOL_VISIBLE
     authority_view
+    : protected detail::parts_base
 {
-    enum
-    {
-        id_user = 0,
-        id_pass,        // leading ':', trailing '@'
-        id_host,
-        id_port,        // leading ':'
-        id_end          // one past the end
-    };
+    detail::url_impl u_;
```

Comment:
:open_mouth: So much simpler

Reply:
in a sense yes, since `authority_view` just recycles the same id enum and struct of offsets and state. Much better than the bulk copy-paste I did originally